<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Getting Started with BLAST on the Command Line | Dr. R. Jordan Price </title> <meta name="author" content="R. Jordan Price"> <meta name="description" content="A guide to help getting started with using the basic functions of BLAST on the command line."> <meta name="keywords" content="bioinformatics, bioinformatician, computational-biology, molecular-biology, functional-genomics, academic-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://rj-price.github.io/blog/2024/blast/"> <script src="/assets/js/theme.js?a5ca4084d3b81624bcfa01156dae2b8e"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> Dr. R. Jordan Price </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/news/">news </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Getting Started with BLAST on the Command Line</h1> <p class="post-meta"> July 15, 2024 </p> <p class="post-tags"> <a href="/blog/2024"> <i class="fa-solid fa-calendar fa-sm"></i> 2024 </a>   ·   <a href="/blog/tag/bioinformatics"> <i class="fa-solid fa-hashtag fa-sm"></i> bioinformatics</a>   <a href="/blog/tag/blast"> <i class="fa-solid fa-hashtag fa-sm"></i> blast</a>     ·   <a href="/blog/category/getting-started"> <i class="fa-solid fa-tag fa-sm"></i> getting-started</a>   </p> </header> <article class="post-content"> <div id="markdown-content"> <p>Bioinformatics relies heavily on sequence identification by comparing nucleotide or protein sequences against known databases. One of the most popular tools for this purpose is BLAST (Basic Local Alignment Search Tool). In this blog post, we’ll explore the basics of using BLAST on the command line, focusing on the two most common types of searches: <code class="language-plaintext highlighter-rouge">blastn</code> and <code class="language-plaintext highlighter-rouge">blastp</code>. We’ll also cover how to perform searches against the NCBI databases, as well as creating and searching a custom local database.</p> <p><br></p> <h2 id="what-is-blast">What is BLAST?</h2> <p>BLAST is a powerful algorithm and software suite from NCBI that allows you to compare a query sequence against a database of sequences to find regions of similarity. This can help identify homologous genes, infer functional and evolutionary relationships, and discover new genes.</p> <p><br></p> <h2 id="types-of-blast-searches">Types of BLAST Searches</h2> <p>There are several types of BLAST searches, the two most common are:</p> <ol> <li> <strong>blastn</strong>: Compares a nucleotide query sequence against a nucleotide sequence database.</li> <li> <strong>blastp</strong>: Compares an amino acid query sequence against a protein sequence database.</li> </ol> <p><br></p> <h2 id="setting-up-blast">Setting Up BLAST</h2> <p>To get started, you’ll need to install the BLAST+ command line tools. You can either download the source files from the <a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/" rel="external nofollow noopener" target="_blank">NCBI BLAST+ download page</a> or install them though <a href="https://anaconda.org/bioconda/blast" rel="external nofollow noopener" target="_blank">conda</a>.</p> <p>If using the source files, make sure that you extract the files and add the BLAST directory to your system’s PATH to use the BLAST commands from any directory.</p> <p><br></p> <h2 id="running-blast-searches-against-ncbi-databases">Running BLAST Searches Against NCBI Databases</h2> <h3 id="1-using-blastn">1. Using <code class="language-plaintext highlighter-rouge">blastn</code> </h3> <p>Let’s start with a <code class="language-plaintext highlighter-rouge">blastn</code> example. Suppose you have a nucleotide sequence in a file called <code class="language-plaintext highlighter-rouge">gene.fasta</code> and you want to identify the closest matching hits to your gene of interest in the NCBI nucleotide database:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>blastn <span class="nt">-query</span> gene.fasta <span class="nt">-remote</span> <span class="nt">-db</span> nt <span class="nt">-out</span> results_blastn.txt <span class="nt">-evalue</span> 1e-6 <span class="nt">-outfmt</span> 6
</code></pre></div></div> <p>This command performs a nucleotide BLAST search of your nucleotide sequence <code class="language-plaintext highlighter-rouge">-query gene.fasta</code> against the NCBI nucleotide database <code class="language-plaintext highlighter-rouge">-remote -db nt</code> and saves the results with an E-value lower than <code class="language-plaintext highlighter-rouge">1e-6</code> in the output file <code class="language-plaintext highlighter-rouge">results_blastn.txt</code> formatted as a table <code class="language-plaintext highlighter-rouge">-outfmt 6</code>.</p> <p><br></p> <h3 id="2-using-blastp">2. Using <code class="language-plaintext highlighter-rouge">blastp</code> </h3> <p>Next, we’ll run a <code class="language-plaintext highlighter-rouge">blastp</code> search with a protein sequence file called <code class="language-plaintext highlighter-rouge">protein.fasta</code> to identify the best hits in the NCBI non-redundant protein database:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>blastp <span class="nt">-query</span> protein.fasta <span class="nt">-remote</span> <span class="nt">-db</span> nr <span class="nt">-out</span> results_blastp.txt <span class="nt">-evalue</span> 1e-6 <span class="nt">-outfmt</span> 6
</code></pre></div></div> <p>This performs a protein BLAST search of your protein sequence <code class="language-plaintext highlighter-rouge">-query protein.fasta</code> against the NCBI non-redundant protein database <code class="language-plaintext highlighter-rouge">-remote -db nr</code> and saves the results with an E-value lower than <code class="language-plaintext highlighter-rouge">1e-6</code> in the output file <code class="language-plaintext highlighter-rouge">results_blastp.txt</code> formatted as a table <code class="language-plaintext highlighter-rouge">-outfmt 6</code>.</p> <p><br></p> <h2 id="creating-and-searching-a-custom-local-database">Creating and Searching a Custom Local Database</h2> <p>Sometimes, you may want to perform searches against a custom database. For example, you may have a known gene that you’d like to identify the homolog of in a <em>de novo</em> assembly. Here’s how you can create and search a local custom BLAST database.</p> <h3 id="1-creating-a-custom-database">1. Creating a Custom Database</h3> <p>First, gather your sequences in a FASTA file, for example, <code class="language-plaintext highlighter-rouge">my_sequences.fasta</code>.</p> <p>To create a nucleotide database:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>makeblastdb <span class="nt">-in</span> my_sequences.fasta <span class="nt">-dbtype</span> nucl <span class="nt">-out</span> my_custom_db
</code></pre></div></div> <p>For a protein database:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>makeblastdb <span class="nt">-in</span> my_sequences.fasta <span class="nt">-dbtype</span> prot <span class="nt">-out</span> my_custom_db
</code></pre></div></div> <p><br></p> <h3 id="2-running-a-blast-search-against-the-custom-database">2. Running a BLAST Search Against the Custom Database</h3> <p>Now, you can run BLAST searches against your custom database. For example, using <code class="language-plaintext highlighter-rouge">blastn</code> with your custom nucleotide database:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>blastn <span class="nt">-query</span> gene.fasta <span class="nt">-db</span> my_custom_db <span class="nt">-out</span> custom_results_blastn.txt <span class="nt">-evalue</span> 1e-6 <span class="nt">-outfmt</span> 6
</code></pre></div></div> <p>Or using <code class="language-plaintext highlighter-rouge">blastp</code> with your custom protein database:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>blastp <span class="nt">-query</span> protein.fasta <span class="nt">-db</span> my_custom_db <span class="nt">-out</span> custom_results_blastp.txt <span class="nt">-evalue</span> 1e-6 <span class="nt">-outfmt</span> 6
</code></pre></div></div> <p><br></p> <h2 id="interpreting-blast-results">Interpreting BLAST Results</h2> <p>BLAST results help determine the potential function of your query sequence by identifying similar sequences in known databases. When using the <code class="language-plaintext highlighter-rouge">-outfmt 6</code> flag to output the results in a tabular format, the data in the columns are represented by the following headers:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore
</code></pre></div></div> <p>1.  <strong>qseqid:</strong> Identifier of the query sequence.</p> <p>2.  <strong>sseqid:</strong> Identifier of the subject sequence that matches your query.</p> <p>3.  <strong>pident:</strong> Percentage of identical matches between the query and subject sequences.</p> <p>4.  <strong>length:</strong> Total alignment length between the query and subject sequences.</p> <p>5.  <strong>mismatch:</strong> Number of positions that are mismatches.</p> <p>6.  <strong>gapopen:</strong> Number of gap openings in the alignment.</p> <p>7.  <strong>qstart:</strong> Start position of the alignment for the query sequence.</p> <p>8.  <strong>qend:</strong> End position of the alignment for the query sequence.</p> <p>9.  <strong>sstart:</strong> Start position of the alignment for the subject sequence.</p> <p>10.  <strong>send:</strong> End position of the alignment for the subject sequence.</p> <p>11.  <strong>evalue:</strong> The number of expected hits of similar quality (lower is better).</p> <p>12.  <strong>bitscore:</strong> The higher the bit-score, the better the sequence similarity.</p> <p><br></p> <h2 id="summary">Summary</h2> <p>BLAST is an essential tool in bioinformatics for sequence identification and comparison. By understanding how to use <code class="language-plaintext highlighter-rouge">blastn</code> and <code class="language-plaintext highlighter-rouge">blastp</code>, and knowing how to create and search a custom database, you can leverage BLAST to gain significant insights into your sequences. Whether you are searching the NCBI databases or your custom datasets, BLAST provides the versatility and power needed for effective sequence analysis.</p> <p>To learn more about the functions and features of the BLAST+ command line tools, take a look at the <a href="https://www.ncbi.nlm.nih.gov/books/NBK279690/" rel="external nofollow noopener" target="_blank">User Manual</a> or use the <code class="language-plaintext highlighter-rouge">-help</code> flag to learn more about the possible arguments for each search.</p> </div> </article> <br> <hr> <br> <ul class="list-disc pl-8"></ul> <h2 class="text-3xl font-semibold mb-4 mt-12">Enjoy Reading This Article?</h2> <p class="mb-2">Here are some more articles you might like to read next:</p> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/bash/">Getting Started with the Bash Command Line</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/jellyfish/">Estimating Genome Size Using Jellyfish and GenomeScope2</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/venn/">4-Way Venn Diagram of Overlapping Gene Expression in R</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/dotfiles/">Useful Bash Aliases and Functions</a> </li> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 R. Jordan Price. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?5d75c11f89cd96294bf5e6dd1ee1bb30"></script> <script defer src="/assets/js/common.js?fcfacfb8c6281f5e68d5a7d348186eb1"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>