<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Dr. R. Jordan Price </title> <meta name="author" content="R. Jordan Price"> <meta name="description" content="Dr. R. Jordan Price, Computational and Molecular Biologist "> <meta name="keywords" content="computational-biology, molecular-biology, functional-genomics, academic-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://rj-price.github.io/"> <script src="/assets/js/theme.js?a5ca4084d3b81624bcfa01156dae2b8e"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item active"> <a class="nav-link" href="/">about <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title"> Dr. R. Jordan Price </h1> <p class="desc">Computational and Molecular Biologist</p> </header> <article> <div class="profile float-right"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/prof_pic-480.webp 480w,/assets/img/prof_pic-800.webp 800w,/assets/img/prof_pic-1400.webp 1400w," sizes="(min-width: 800px) 231.0px, (min-width: 576px) 30vw, 95vw" type="image/webp"> <img src="/assets/img/prof_pic.jpg?05cabc358aea8a31b8ab5616efeea577" class="img-fluid z-depth-1 rounded" width="100%" height="auto" alt="prof_pic.jpg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> <div class="more-info"> <p>NIAB</p> <p>Cambridge, UK</p> </div> </div> <div class="clearfix"> <p>Hi! Iâ€™m Jordan ðŸ‘‹</p> <ul> <li>Iâ€™m a wet lab molecular biologist turned bioinformatician.</li> <li>I am currently a junior group leader at NIAB in Cambridge, UK.</li> <li>My research focusses on functional genomics and biotechnology of fungi important for food, viticulture and agriculture.</li> <li>I use a combination of molecular and computational approaches to understand the links between genome evolution and biological traits.</li> <li>I also occasionally dabble in plant genomics.</li> </ul> </div> <h2> <a href="/news/" style="color: inherit">news</a> </h2> <div class="news"> <div class="table-responsive" style="max-height: 60vw"> <table class="table table-sm table-borderless"> <tr> <th scope="row" style="width: 20%">Jul 02, 2024</th> <td> <a class="news-title" href="/news/2024-07-02/">Strawberry transformation pre-print published</a> </td> </tr> <tr> <th scope="row" style="width: 20%">Apr 13, 2024</th> <td> <a class="news-title" href="/news/2024-04-13/">Fungal comparative genomics pre-print published</a> </td> </tr> <tr> <th scope="row" style="width: 20%">Apr 08, 2024</th> <td> Attended the <a href="https://microbiologysociety.org/" rel="external nofollow noopener" target="_blank">Microbiology Society</a> Annual Conference 2024 in Edinburgh, and chaired two sessions that I had helped to organise as part of the Eukaryotic Division. </td> </tr> <tr> <th scope="row" style="width: 20%">Feb 28, 2024</th> <td> Invited to act as External Examiner for University of Nottingham PhD student <a href="https://eprints.nottingham.ac.uk/78053/" rel="external nofollow noopener" target="_blank">Kamil Szepe</a>, who passed with minor corrections following an enjoyable discussion around his work on improving the amino acid profile of mycoprotein. </td> </tr> <tr> <th scope="row" style="width: 20%">Jan 24, 2024</th> <td> Attended <a href="https://festivalofgenomics.com/london/en/page/home" rel="external nofollow noopener" target="_blank">The Festival of Genomics &amp; Biodata</a> in London, a fascinating insight into the diverse and innovative research happening in the genomics and multi-omics fields. </td> </tr> </table> </div> </div> <h2> <a href="/publications/" style="color: inherit">selected publications</a> </h2> <div class="publications"> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 abbr"> </div> <div id="Bates2024.04.11.589035" class="col-sm-8"> <div class="title">Comparative genomics and transcriptomics reveal differences in effector complement and expression between races of Fusarium oxysporum f.sp. lactucae</div> <div class="author"> Helen J. Bates ,Â Jamie Pike ,Â <em>R. Jordan Price</em>,Â Sascha Jenkins ,Â John Connell , and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Andrew Legg, Andrew Armitage, Richard J. Harrison, John P. Clarkson' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>bioRxiv</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>This study presents the first genome and transcriptome analyses for Fusarium oxysporum f.sp. lactucae (Fola) which causes Fusarium wilt disease of lettuce. Long-read genome sequencing of three race 1 (Fola1) and three race 4 (Fola4) isolates revealed key differences in putative effector complement between races and with other F. oxysporum f.spp. following mimp-based bioinformatic analyses. Notably, homologues of Secreted in Xylem (SIX) genes, also present in many other F. oxysporum f.spp, were identified in Fola, with both SIX9 and SIX14 (multiple copies with sequence variants) present in both Fola1 and Fola4. All Fola4 isolates also contained an additional single copy of SIX8. RNAseq of lettuce following infection with Fola1 and Fola4 isolates identified highly expressed effectors, some of which were homologues of those reported in other F. oxysporum f.spp. including several in F. oxysporum f.sp. apii. Although SIX8, SIX9 and SIX14 were all highly expressed in Fola4, of the two SIX genes present in Fola1, only SIX9 was expressed as further analysis revealed that copies of SIX14 gene copies were disrupted by insertion of a transposable element. Two variants of Fola4 were also identified based on different genome and effector-based analyses. This included two different SIX8 sequence variants which were divergently transcribed from a shared promoter with either PSE1 or PSL1 respectively. In addition there was evidence of two independent instances of HCT in the different Fola4 variants. The involvement of helitrons in Fola genome rearrangement and gene expression is discussed.Competing Interest StatementThe authors have declared no competing interest.The datasets generated for this study can be found in the article and supplementary material. All raw sequencing data and new genome assemblies presented have been submitted to the NCBI under the BioProject ID PRJNA1092066. Further enquiries can be directed to the corresponding author.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> </div> <div id="10.1371/journal.pone.0285756" class="col-sm-8"> <div class="title">Chromosome-scale genome sequence assemblies of the â€˜Autumn Blissâ€™ and â€˜Malling Jewelâ€™ cultivars of the highly heterozygous red raspberry (Rubus idaeus L.) derived from long-read Oxford Nanopore sequence data</div> <div class="author"> <em>R. Jordan Price</em>,Â Jahn Davik ,Â Felicidad FernandÃ©z FernandÃ©z ,Â Helen J. Bates ,Â Samantha Lynn , and <span class="more-authors" title="click to view 7 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '7 more authors' ? 'Charlotte F. Nellist, Matteo Buti, Dag RÃ¸en, Nada Å urbanovski, Muath Alsheikh, Richard J. Harrison, Daniel James Sargent' : '7 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">7 more authors</span> </div> <div class="periodical"> <em>PLOS ONE</em>, May 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>Red raspberry (Rubus idaeus L.) is an economically valuable soft-fruit species with a relatively small (Â 300 Mb) but highly heterozygous diploid (2n = 2x = 14) genome. Chromosome-scale genome sequences are a vital tool in unravelling the genetic complexity controlling traits of interest in crop plants such as red raspberry, as well as for functional genomics, evolutionary studies, and pan-genomics diversity studies. In this study, we developed genome sequences of a primocane fruiting variety (â€˜Autumn Blissâ€™) and a floricane variety (â€˜Malling Jewelâ€™). The use of long-read Oxford Nanopore Technologies sequencing data yielded long read lengths that permitted well resolved genome sequences for the two cultivars to be assembled. The de novo assemblies of â€˜Malling Jewelâ€™ and â€˜Autumn Blissâ€™ contained 79 and 136 contigs respectively, and 263.0 Mb of the â€˜Autumn Blissâ€™ and 265.5 Mb of the â€˜Malling Jewelâ€™ assembly could be anchored unambiguously to a previously published red raspberry genome sequence of the cultivar â€˜Anitraâ€™. Single copy ortholog analysis (BUSCO) revealed high levels of completeness in both genomes sequenced, with 97.4% of sequences identified in â€˜Autumn Blissâ€™ and 97.7% in â€˜Malling Jewelâ€™. The density of repetitive sequence contained in the â€˜Autumn Blissâ€™ and â€˜Malling Jewelâ€™ assemblies was significantly higher than in the previously published assembly and centromeric and telomeric regions were identified in both assemblies. A total of 42,823 protein coding regions were identified in the â€˜Autumn Blissâ€™ assembly, whilst 43,027 were identified in the â€˜Malling Jewelâ€™ assembly. These chromosome-scale genome sequences represent an excellent genomics resource for red raspberry, particularly around the highly repetitive centromeric and telomeric regions of the genome that are less complete in the previously published â€˜Anitraâ€™ genome sequence.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> </div> <div id="10.1371/journal.pgen.1010576" class="col-sm-8"> <div class="title">Stress combined with loss of the Candida albicans SUMO protease Ulp2 triggers selection of aneuploidy via a two-step process</div> <div class="author"> Marzia Rizzo ,Â Natthapon Soisangwan ,Â Samuel Vega-Estevez ,Â <em>R. Jordan Price</em>,Â Chloe Uyl , and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Elise Iracane, Matt Shaw, Jan Soetaert, Anna Selmecki, Alessia Buscaino' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">5 more authors</span> </div> <div class="periodical"> <em>PLOS Genetics</em>, Dec 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>A delicate balance between genome stability and instability ensures genome integrity while generating genetic diversity, a critical step for evolution. Indeed, while excessive genome instability is harmful, moderated genome instability can drive adaptation to novel environments by maximising genetic variation. Candida albicans, a human fungal pathogen that colonises different parts of the human body, adapts rapidly and frequently to different hostile host microenvironments. In this organism, the ability to generate large-scale genomic variation is a key adaptative mechanism triggering dangerous infections even in the presence of antifungal drugs. Understanding how fitter novel karyotypes are selected is key to determining how C. albicans and other microbial pathogens establish infections. Here, we identified the SUMO protease Ulp2 as a regulator of C. albicans genome integrity through genetic screening. Deletion of ULP2 leads to increased genome instability, enhanced genome variation and reduced fitness in the absence of additional stress. The combined stress caused by the lack of ULP2 and antifungal drug treatment leads to the selection of adaptive segmental aneuploidies that partially rescue the fitness defects of ulp2Î”/Î” cells. Short and long-read genomic sequencing demonstrates that these novel genotypes are selected via a two-step process leading to the formation of novel chromosomal fragments with breakpoints at microhomology regions and DNA repeats.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> </div> <div id="doi:10.1128/mbio.01376-19" class="col-sm-8"> <div class="title">Chromatin Profiling of the Repetitive and Nonrepetitive Genomes of the Human Fungal Pathogen Candida albicans</div> <div class="author"> <em>R. Jordan Price</em>,Â Esther Weindling ,Â Judith Berman ,Â andÂ Alessia Buscaino </div> <div class="periodical"> <em>mBio</em>, Dec 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p> The fungus Candida albicans is an opportunistic pathogen that normally lives on the human body without causing any harm. However, C. albicans is also a dangerous pathogen responsible for millions of infections annually. C. albicans is such a successful pathogen because it can adapt to and thrive in different environments. Chemical modifications of chromatin, the structure that packages DNA into cells, can allow environmental adaptation by regulating gene expression and genome organization. Surprisingly, the contribution of chromatin modification to C. albicans biology is still largely unknown. For the first time, we analyzed C. albicans chromatin modifications on a genome-wide basis. We demonstrate that specific chromatin states are associated with distinct regions of the C. albicans genome and identify the roles of the chromatin modifiers Sir2 and Set1 in shaping C. albicans chromatin and gene expression. Eukaryotic genomes are packaged into chromatin structures that play pivotal roles in regulating all DNA-associated processes. Histone posttranslational modifications modulate chromatin structure and function, leading to rapid regulation of gene expression and genome stability, key steps in environmental adaptation. Candida albicans, a prevalent fungal pathogen in humans, can rapidly adapt and thrive in diverse host niches. The contribution of chromatin to C. albicans biology is largely unexplored. Here, we generated the first comprehensive chromatin profile of histone modifications (histone H3 trimethylated on lysine 4 [H3K4me3], histone H3 acetylated on lysine 9 [H3K9Ac], acetylated lysine 16 on histone H4 [H4K16Ac], and Î³H2A) across the C. albicans genome and investigated its relationship to gene expression by harnessing genome-wide sequencing approaches. We demonstrated that gene-rich nonrepetitive regions are packaged into canonical euchromatin in association with histone modifications that mirror their transcriptional activity. In contrast, repetitive regions are assembled into distinct chromatin states; subtelomeric regions and the ribosomal DNA (rDNA) locus are assembled into heterochromatin, while major repeat sequences and transposons are packaged in chromatin that bears features of euchromatin and heterochromatin. Genome-wide mapping of Î³H2A, a marker of genome instability, identified potential recombination-prone genomic loci. Finally, we present the first quantitative chromatin profiling in C. albicans to delineate the role of the chromatin modifiers Sir2 and Set1 in controlling chromatin structure and gene expression. This report presents the first genome-wide chromatin profiling of histone modifications associated with the C. albicans genome. These epigenomic maps provide an invaluable resource to understand the contribution of chromatin to C. albicans biology and identify aspects of C. albicans chromatin organization that differ from that of other yeasts. IMPORTANCE The fungus Candida albicans is an opportunistic pathogen that normally lives on the human body without causing any harm. However, C. albicans is also a dangerous pathogen responsible for millions of infections annually. C. albicans is such a successful pathogen because it can adapt to and thrive in different environments. Chemical modifications of chromatin, the structure that packages DNA into cells, can allow environmental adaptation by regulating gene expression and genome organization. Surprisingly, the contribution of chromatin modification to C. albicans biology is still largely unknown. For the first time, we analyzed C. albicans chromatin modifications on a genome-wide basis. We demonstrate that specific chromatin states are associated with distinct regions of the C. albicans genome and identify the roles of the chromatin modifiers Sir2 and Set1 in shaping C. albicans chromatin and gene expression. </p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> </div> <div id="Freire-BenÃ©itez2016" class="col-sm-8"> <div class="title">Candida albicans repetitive elements display epigenetic diversity and plasticity</div> <div class="author"> VerÃ³nica Freire-BenÃ©itez ,Â <em>R. Jordan Price</em>,Â Daniel Tarrant ,Â Judith Berman ,Â andÂ Alessia Buscaino </div> <div class="periodical"> <em>Scientific Reports</em>, Mar 2016 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>Transcriptionally silent heterochromatin is associated with repetitive DNA. It is poorly understood whether and how heterochromatin differs between different organisms and whether its structure can be remodelled in response to environmental signals. Here, we address this question by analysing the chromatin state associated with DNA repeats in the human fungal pathogen Candida albicans. Our analyses indicate that, contrary to model systems, each type of repetitive element is assembled into a distinct chromatin state. Classical Sir2-dependent hypoacetylated and hypomethylated chromatin is associated with the rDNA locus while telomeric regions are assembled into a weak heterochromatin that is only mildly hypoacetylated and hypomethylated. Major Repeat Sequences, a class of tandem repeats, are assembled into an intermediate chromatin state bearing features of both euchromatin and heterochromatin. Marker gene silencing assays and genome-wide RNA sequencing reveals that C. albicans heterochromatin represses expression of repeat-associated coding and non-coding RNAs. We find that telomeric heterochromatin is dynamic and remodelled upon an environmental change. Weak heterochromatin is associated with telomeres at 30\thinspace\textdegreeC, while robust heterochromatin is assembled over these regions at 39\thinspace\textdegreeC, a temperature mimicking moderate fever in the host. Thus in C. albicans, differential chromatin states controls gene expression and epigenetic plasticity is linked to adaptation.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 abbr"> </div> <div id="PRICE2015532" class="col-sm-8"> <div class="title">Folic acid supplementation in vitro induces cell typeâ€“specific changes in BRCA1 and BRCA 2 mRNA expression, but does not alter DNA methylation of their promoters or DNA repair</div> <div class="author"> <em>R. Jordan Price</em>,Â Karen A. Lillycrop ,Â andÂ Graham C. Burdge </div> <div class="periodical"> <em>Nutrition Research</em>, Mar 2015 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>Dietary supplementation with folic acid (FA) has been shown to induce opposing effects on cancer-related outcomes. The mechanism underlying such heterogeneity is unclear. We hypothesized that FA supplementation induces changes in breast cancerâ€“associated (BRCA) genes 1 and 2 expression and function through altered epigenetic regulation in a cell typeâ€“dependent manner. We investigated the effect of treating normal and cancer cells with physiologically relevant FA concentrations on the mRNA and protein expression, capacity for DNA repair, and DNA methylation of BRCA1 and BRCA2. FA treatment induced dose-related increases in BRCA1 mRNA expression in HepG2, Huh-7D12, Hs578T, and JURKAT and in BRCA2 in HepG2, Hs578T, MCF7, and MDA-MB-157 cells. FA did not affect the corresponding normal cells or on any of the ovarian cell lines. Folic acid induced increased BRCA1 protein expression in Hs578T, but not HepG2 cells, whereas BRCA2 protein levels were undetectable. FA treatment did not alter DNA repair in liver-derived cells, whereas there were transient effects on breast-derived cells. There was no effect of FA treatment on BRCA1 or BRCA2 DNA methylation, although there was some variation in the methylation of specific CpG loci between some cell lines. Overall, these findings show that the effects of FA on BRCA-related outcomes differ between cells lines, but the biological consequences of induced changes in BRCA expression appear to be at most limited.</p> </div> </div> </div> </li> </ol> </div> <div class="social"> <div class="contact-icons"> <a href="mailto:%6A%6F%72%64%61%6E.%70%72%69%63%65@%6E%69%61%62.%63%6F%6D" title="email"><i class="fa-solid fa-envelope"></i></a> <a href="https://scholar.google.com/citations?user=Jc2K0ZQAAAAJ" title="Google Scholar" rel="external nofollow noopener" target="_blank"><i class="ai ai-google-scholar"></i></a> <a href="https://github.com/rj-price" title="GitHub" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-github"></i></a> <a href="https://www.linkedin.com/in/jordan-price-95429784" title="LinkedIn" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-linkedin"></i></a> <a href="https://twitter.com/DrRJordanPrice" title="X" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-x-twitter"></i></a> <a href="/feed.xml" title="RSS Feed"><i class="fa-solid fa-square-rss"></i></a> </div> <div class="contact-note"></div> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> Â© Copyright 2024 R. Jordan Price. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?5d75c11f89cd96294bf5e6dd1ee1bb30"></script> <script defer src="/assets/js/common.js?fcfacfb8c6281f5e68d5a7d348186eb1"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>