<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Dr. R. Jordan Price </title> <meta name="author" content="R. Jordan Price"> <meta name="description" content="Dr. R. Jordan Price, Bioinformatician and Molecular Biologist "> <meta name="keywords" content="bioinformatics, bioinformatician, computational-biology, molecular-biology, functional-genomics, academic-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://rj-price.github.io/"> <script src="/assets/js/theme.js?a5ca4084d3b81624bcfa01156dae2b8e"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item active"> <a class="nav-link" href="/">about <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/news/">news </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/apps/">apps </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title"> Dr. R. Jordan Price </h1> <p class="desc">Bioinformatician and Molecular Biologist</p> </header> <article> <div class="profile float-right"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/prof_pic-480.webp 480w,/assets/img/prof_pic-800.webp 800w,/assets/img/prof_pic-1400.webp 1400w," sizes="(min-width: 800px) 231.0px, (min-width: 576px) 30vw, 95vw" type="image/webp"> <img src="/assets/img/prof_pic.jpg?05cabc358aea8a31b8ab5616efeea577" class="img-fluid z-depth-1 rounded" width="100%" height="auto" alt="prof_pic.jpg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> <div class="more-info"> <p>NIAB</p> <p>Cambridge, UK</p> </div> </div> <div class="clearfix"> <p>Hi! I’m Jordan 👋</p> <ul> <li>I’m a molecular biologist turned bioinformatician.</li> <li>I am currently a junior group leader at NIAB in Cambridge, UK.</li> <li>My research focusses on functional genomics and biotechnology of fungi important for food, viticulture and agriculture.</li> <li>I use a combination of molecular and computational approaches to understand the links between genome evolution and biological traits.</li> <li>I also dabble in plant genomics.</li> </ul> <p>Currently learning:</p> <ul> <li>Nextflow pipeline development</li> <li>Running local LLMs with RAG</li> </ul> </div> <h2> <a href="/news/" style="color: inherit">news</a> </h2> <div class="news"> <div class="table-responsive" style="max-height: 60vw"> <table class="table table-sm table-borderless"> <tr> <th scope="row" style="width: 20%">Jun 30, 2025</th> <td> University of Warwick student Lowri Wigley starts her <a href="https://www.bspp.org.uk/funds/vacation/" rel="external nofollow noopener" target="_blank">BSPP Undergraduate Summer Studentship</a> investigating the presence of Helitron transposable elements in lettuce-infecting <em>Fusarium oxysporum</em>. </td> </tr> <tr> <th scope="row" style="width: 20%">Jun 06, 2025</th> <td> An advance version of our review paper, “<a href="https://doi.org/10.1146/annurev-cellbio-101123-093247" rel="external nofollow noopener" target="_blank">Getting to the Route: The Evolution of Nitrogen-Fixing Nodules</a>” has just been published in Annual Review of Cell and Developmental Biology. </td> </tr> <tr> <th scope="row" style="width: 20%">May 30, 2025</th> <td> Congratulations to PhD student Finlay Bourquin who has submitted her thesis titled “Understanding resistance to <em>Botrytis cinerea</em> in strawberries”. </td> </tr> <tr> <th scope="row" style="width: 20%">Apr 30, 2025</th> <td> This week I have Rima Gnaim from the <a href="https://rlalab.org/" rel="external nofollow noopener" target="_blank">Ledesma-Amaro lab</a> visiting to learn all about <em>F. venenatum</em> transformation. </td> </tr> <tr> <th scope="row" style="width: 20%">Apr 01, 2025</th> <td> Attended the <a href="https://microbiologysociety.org/" rel="external nofollow noopener" target="_blank">Microbiology Society</a> Annual Conference 2025 in Liverpool, and chaired the “From Microbes to Meals” session that I helped to organise as part of the Eukaryotic Division. </td> </tr> </table> </div> </div> <h2> <a href="/publications/" style="color: inherit">selected publications</a> </h2> <div class="publications"> <ol class="bibliography"> <li> <div class="row"> <div id="Connell2025" class="col-sm-10"> <div class="title">Mutation of the LRG1 Rho-GAP gene is responsible for the hyper branching C-variant phenotype in the quorn mycoprotein fungus Fusarium venenatum A3/5</div> <div class="author"> John Connell , Helen J. Bates , Ivey Geoghegan , Fiona Wilson , Richard J. Harrison , and <em>R. Jordan Price</em> </div> <div class="periodical"> <em>Fungal Biology and Biotechnology</em>, 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1186/s40694-025-00195-8" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Connell_et_al_2025.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>Quorn mycoprotein, a protein-rich meat alternative, is produced through large-scale fermentation of the fungus Fusarium venenatum. However, a major challenge during F. venenatum fermentation is the consistent appearance of mutants called colonial variants (C-variants). These C-variants have a highly branched morphology, which ultimately lead to a less desirable final product and early termination of the fermentation process. This study aimed to identify the genetic mutations responsible for C-variant morphology.</p> </div> </div> </div> </li> <li> <div class="row"> <div id="10.1371/journal.pone.0318692" class="col-sm-10"> <div class="title">A CACTA-like transposon in the Anthocyanidin synthase 1 (Ans-1) gene is responsible for apricot fruit colour in the raspberry (Rubus idaeus) cultivar ‘Varnes’</div> <div class="author"> Daniel James Sargent , Matteo Buti , Stefan Martens , Claudio Pugliesi , Kjersti Aaby , Dag Røen , Chandra Bhan Yadav , Felicidad Fernández Fernández , Muath Alsheikh , Jahn Davik , and <em>R. Jordan Price</em> </div> <div class="periodical"> <em>PLOS ONE</em>, Feb 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1371/journal.pone.0318692" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Sargent_et_al_2025.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>Cultivated raspberries (Rubus idaeus L.) most commonly bear small, red, highly aromatic fruits. Their colour is derived predominantly from anthocyanins, water soluble polyphenolic pigments, but as well as red forms, there exist cultivars that display yellow- and apricot-coloured fruits. In this investigation, we used a multi-omics approach to elucidate the genetic basis of the apricot fruit colour in raspberry. Using metabolomics, we quantified anthocyanins in red and apricot raspberry fruits and demonstrated that, in contrast to red-fruited raspberries, fruits of the apricot cultivar ‘Varnes’ contain low concentrations of only a small number of anthocyanin compounds. By performing RNASeq, we revealed differential expression patterns in the apricot-fruited ‘Varnes’ for genes in the anthocyanin biosynthesis pathway and following whole genome sequencing using long-read Oxford Nanopore Technologies sequencing, we identified a CACTA-like transposable element (TE) in the second exon of the Anthocyanidin synthase (Ans) gene that caused a truncated predicted ANS protein. PCR confirmed the presence in heterozygous form of the transposon in an unrelated, red-fruited cultivar ‘Veten’, indicating apricot fruit colour is recessive to red and that it may be widespread in raspberry germplasm, potentially explaining why apricot forms appear at regular intervals in modern raspberry breeding populations.</p> </div> </div> </div> </li> <li> <div class="row"> <div id="10.3389/fpls.2024.1415534" class="col-sm-10"> <div class="title">Comparative genomics and transcriptomics reveal differences in effector complement and expression between races of Fusarium oxysporum f.sp. lactucae</div> <div class="author"> Helen J. Bates , Jamie Pike , <em>R. Jordan Price</em>, Sascha Jenkins , John Connell , Andrew Legg , Andrew Armitage , Richard J. Harrison , and John P. Clarkson </div> <div class="periodical"> <em>Frontiers in Plant Science</em>, Oct 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.frontiersin.org/journals/plant-science/articles/10.3389/fpls.2024.1415534" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Bates_et_al_2024.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p></p> <p>This study presents the first genome and transcriptome analyses for <italic>Fusarium oxysporum</italic> f. sp. <italic>lactucae</italic> (Fola) which causes Fusarium wilt disease of lettuce. Long-read genome sequencing of three race 1 (Fola1) and three race 4 (Fola4) isolates revealed key differences in putative effector complement between races and with other <italic>F. oxysporum</italic> ff. spp. following <italic>mimp</italic>-based bioinformatic analyses. Notably, homologues of <italic>Secreted in Xylem</italic> (<italic>SIX</italic>) genes, also present in many other <italic>F. oxysporum</italic> ff. spp, were identified in Fola, with both <italic>SIX9</italic> and <italic>SIX14</italic> (multiple copies with sequence variants) present in both Fola1 and Fola4. All Fola4 isolates also contained an additional single copy of <italic>SIX8</italic>. RNAseq of lettuce following infection with Fola1 and Fola4 isolates identified highly expressed effectors, some of which were homologues of those reported in other <italic>F. oxysporum</italic> ff. spp. including several in <italic>F. oxysporum</italic> f. sp. <italic>apii</italic>. Although <italic>SIX8</italic>, <italic>SIX9</italic> and <italic>SIX14</italic> were all highly expressed in Fola4, of the two <italic>SIX</italic> genes present in Fola1, only <italic>SIX9</italic> was expressed as further analysis revealed that <italic>SIX14</italic> gene copies were disrupted by insertion of a transposable element. Two variants of Fola4 were also identified based on different genome and effector-based analyses. This included two different <italic>SIX8</italic> sequence variants which were divergently transcribed from a shared promoter with either <italic>PSE1</italic> or <italic>PSL1</italic> respectively. In addition, there was evidence of two independent instances of HCT in the different Fola4 variants. The involvement of helitrons in Fola genome rearrangement and gene expression is discussed.</p> </div> </div> </div> </li> <li> <div class="row"> <div id="10.1371/journal.pone.0285756" class="col-sm-10"> <div class="title">Chromosome-scale genome sequence assemblies of the ‘Autumn Bliss’ and ‘Malling Jewel’ cultivars of the highly heterozygous red raspberry (Rubus idaeus L.) derived from long-read Oxford Nanopore sequence data</div> <div class="author"> <em>R. Jordan Price</em>, Jahn Davik , Felicidad Fernandéz Fernandéz , Helen J. Bates , Samantha Lynn , Charlotte F. Nellist , Matteo Buti , Dag Røen , Nada Šurbanovski , Muath Alsheikh , Richard J. Harrison , and Daniel James Sargent </div> <div class="periodical"> <em>PLOS ONE</em>, May 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1371/journal.pone.0285756" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Price_et_al_2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>Red raspberry (Rubus idaeus L.) is an economically valuable soft-fruit species with a relatively small ( 300 Mb) but highly heterozygous diploid (2n = 2x = 14) genome. Chromosome-scale genome sequences are a vital tool in unravelling the genetic complexity controlling traits of interest in crop plants such as red raspberry, as well as for functional genomics, evolutionary studies, and pan-genomics diversity studies. In this study, we developed genome sequences of a primocane fruiting variety (‘Autumn Bliss’) and a floricane variety (‘Malling Jewel’). The use of long-read Oxford Nanopore Technologies sequencing data yielded long read lengths that permitted well resolved genome sequences for the two cultivars to be assembled. The de novo assemblies of ‘Malling Jewel’ and ‘Autumn Bliss’ contained 79 and 136 contigs respectively, and 263.0 Mb of the ‘Autumn Bliss’ and 265.5 Mb of the ‘Malling Jewel’ assembly could be anchored unambiguously to a previously published red raspberry genome sequence of the cultivar ‘Anitra’. Single copy ortholog analysis (BUSCO) revealed high levels of completeness in both genomes sequenced, with 97.4% of sequences identified in ‘Autumn Bliss’ and 97.7% in ‘Malling Jewel’. The density of repetitive sequence contained in the ‘Autumn Bliss’ and ‘Malling Jewel’ assemblies was significantly higher than in the previously published assembly and centromeric and telomeric regions were identified in both assemblies. A total of 42,823 protein coding regions were identified in the ‘Autumn Bliss’ assembly, whilst 43,027 were identified in the ‘Malling Jewel’ assembly. These chromosome-scale genome sequences represent an excellent genomics resource for red raspberry, particularly around the highly repetitive centromeric and telomeric regions of the genome that are less complete in the previously published ‘Anitra’ genome sequence.</p> </div> </div> </div> </li> <li> <div class="row"> <div id="doi:10.1128/mbio.01376-19" class="col-sm-10"> <div class="title">Chromatin Profiling of the Repetitive and Nonrepetitive Genomes of the Human Fungal Pathogen Candida albicans</div> <div class="author"> <em>R. Jordan Price</em>, Esther Weindling , Judith Berman , and Alessia Buscaino </div> <div class="periodical"> <em>mBio</em>, May 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://journals.asm.org/doi/abs/10.1128/mbio.01376-19" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/Price_et_al_2019.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p> The fungus Candida albicans is an opportunistic pathogen that normally lives on the human body without causing any harm. However, C. albicans is also a dangerous pathogen responsible for millions of infections annually. C. albicans is such a successful pathogen because it can adapt to and thrive in different environments. Chemical modifications of chromatin, the structure that packages DNA into cells, can allow environmental adaptation by regulating gene expression and genome organization. Surprisingly, the contribution of chromatin modification to C. albicans biology is still largely unknown. For the first time, we analyzed C. albicans chromatin modifications on a genome-wide basis. We demonstrate that specific chromatin states are associated with distinct regions of the C. albicans genome and identify the roles of the chromatin modifiers Sir2 and Set1 in shaping C. albicans chromatin and gene expression. Eukaryotic genomes are packaged into chromatin structures that play pivotal roles in regulating all DNA-associated processes. Histone posttranslational modifications modulate chromatin structure and function, leading to rapid regulation of gene expression and genome stability, key steps in environmental adaptation. Candida albicans, a prevalent fungal pathogen in humans, can rapidly adapt and thrive in diverse host niches. The contribution of chromatin to C. albicans biology is largely unexplored. Here, we generated the first comprehensive chromatin profile of histone modifications (histone H3 trimethylated on lysine 4 [H3K4me3], histone H3 acetylated on lysine 9 [H3K9Ac], acetylated lysine 16 on histone H4 [H4K16Ac], and γH2A) across the C. albicans genome and investigated its relationship to gene expression by harnessing genome-wide sequencing approaches. We demonstrated that gene-rich nonrepetitive regions are packaged into canonical euchromatin in association with histone modifications that mirror their transcriptional activity. In contrast, repetitive regions are assembled into distinct chromatin states; subtelomeric regions and the ribosomal DNA (rDNA) locus are assembled into heterochromatin, while major repeat sequences and transposons are packaged in chromatin that bears features of euchromatin and heterochromatin. Genome-wide mapping of γH2A, a marker of genome instability, identified potential recombination-prone genomic loci. Finally, we present the first quantitative chromatin profiling in C. albicans to delineate the role of the chromatin modifiers Sir2 and Set1 in controlling chromatin structure and gene expression. This report presents the first genome-wide chromatin profiling of histone modifications associated with the C. albicans genome. These epigenomic maps provide an invaluable resource to understand the contribution of chromatin to C. albicans biology and identify aspects of C. albicans chromatin organization that differ from that of other yeasts. IMPORTANCE The fungus Candida albicans is an opportunistic pathogen that normally lives on the human body without causing any harm. However, C. albicans is also a dangerous pathogen responsible for millions of infections annually. C. albicans is such a successful pathogen because it can adapt to and thrive in different environments. Chemical modifications of chromatin, the structure that packages DNA into cells, can allow environmental adaptation by regulating gene expression and genome organization. Surprisingly, the contribution of chromatin modification to C. albicans biology is still largely unknown. For the first time, we analyzed C. albicans chromatin modifications on a genome-wide basis. We demonstrate that specific chromatin states are associated with distinct regions of the C. albicans genome and identify the roles of the chromatin modifiers Sir2 and Set1 in shaping C. albicans chromatin and gene expression. </p> </div> </div> </div> </li> </ol> </div> <div class="social"> <div class="contact-icons"> <a href="mailto:%6A%6F%72%64%61%6E.%70%72%69%63%65@%6E%69%61%62.%63%6F%6D" title="email"><i class="fa-solid fa-envelope"></i></a> <a href="https://scholar.google.com/citations?user=Jc2K0ZQAAAAJ" title="Google Scholar" rel="external nofollow noopener" target="_blank"><i class="ai ai-google-scholar"></i></a> <a href="https://github.com/rj-price" title="GitHub" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-github"></i></a> <a href="https://www.linkedin.com/in/jordan-price-95429784" title="LinkedIn" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-linkedin"></i></a> <a href="/feed.xml" title="RSS Feed"><i class="fa-solid fa-square-rss"></i></a> </div> <div class="contact-note"></div> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 R. Jordan Price. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?5d75c11f89cd96294bf5e6dd1ee1bb30"></script> <script defer src="/assets/js/common.js?fcfacfb8c6281f5e68d5a7d348186eb1"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>