<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> 4-Way Venn Diagram of Overlapping Gene Expression in R | Dr. R. Jordan Price </title> <meta name="author" content="R. Jordan Price"> <meta name="description" content="A guide to visualising the overlap between differentially expressed genes (DEGs) in R."> <meta name="keywords" content="bioinformatics, bioinformatician, computational-biology, molecular-biology, functional-genomics, academic-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://rj-price.github.io/blog/2024/venn/"> <script src="/assets/js/theme.js?a5ca4084d3b81624bcfa01156dae2b8e"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> Dr. R. Jordan Price </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/news/">news </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">4-Way Venn Diagram of Overlapping Gene Expression in R</h1> <p class="post-meta"> October 07, 2024 </p> <p class="post-tags"> <a href="/blog/2024"> <i class="fa-solid fa-calendar fa-sm"></i> 2024 </a>   ·   <a href="/blog/tag/bioinformatics"> <i class="fa-solid fa-hashtag fa-sm"></i> bioinformatics</a>   <a href="/blog/tag/rnaseq"> <i class="fa-solid fa-hashtag fa-sm"></i> rnaseq</a>   <a href="/blog/tag/visualisation"> <i class="fa-solid fa-hashtag fa-sm"></i> visualisation</a>   <a href="/blog/tag/venn"> <i class="fa-solid fa-hashtag fa-sm"></i> venn</a>     ·   <a href="/blog/category/getting-started"> <i class="fa-solid fa-tag fa-sm"></i> getting-started</a>   </p> </header> <article class="post-content"> <div id="markdown-content"> <p>Visualising the overlap between differentially expressed genes (DEGs) across conditions or species is key to uncovering shared biological pathways from transcriptomics data.</p> <p>In our recent study published in <a href="https://plantmethods.biomedcentral.com/articles/10.1186/s13007-024-01270-8" rel="external nofollow noopener" target="_blank"><strong>Plant Methods</strong></a>, we analysed RNAseq data to investigate the impact of GRF-GIF developmental regulator chimeras on gene expression in the strawberry model organism <em>Fragaria vesca</em>.</p> <p>In this blog post, we’ll take a quick walk through how I created the 4-way Venn diagram (seen in <a href="https://plantmethods.biomedcentral.com/articles/10.1186/s13007-024-01270-8/figures/4" rel="external nofollow noopener" target="_blank">Figure 4c</a>) that compares DEGs between four different lines using R.</p> <p><br></p> <h2 id="data">Data</h2> <p>Our analysis focussed on DEGs from <em>F. vesca</em> lines transformed with GRF-GIF chimeras from citrus (Cc GRF-GIF), grape (Vv GRF-GIF), grape with microRNA perturbation (Vv miR GRF-GIF), and wheat (Ta GRF-GIF).</p> <p>Following differential expression testing using DESeq2, I exported the DEG names for each line to a new file, imported these into R and created a list of the data for the Venn diagram.</p> <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Import DEG gene names</span><span class="w">
</span><span class="n">citris</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readLines</span><span class="p">(</span><span class="s2">"GRF-GIF_Citris_vs_Control_SigDEGs_names.txt"</span><span class="p">)</span><span class="w">
</span><span class="n">vitis</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readLines</span><span class="p">(</span><span class="s2">"GRF-GIF_Vitis_vs_Control_SigDEGs_names.txt"</span><span class="p">)</span><span class="w">
</span><span class="n">vitis_miR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readLines</span><span class="p">(</span><span class="s2">"GRF-GIF_Vitis_miR_vs_Control_SigDEGs_names.txt"</span><span class="p">)</span><span class="w">
</span><span class="n">wheat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readLines</span><span class="p">(</span><span class="s2">"GRF-GIF_Wheat_vs_Control_SigDEGs_names.txt"</span><span class="p">)</span><span class="w">

</span><span class="c1"># Create list of all data</span><span class="w">
</span><span class="n">venndata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="w">
  </span><span class="s2">"Cc GRF-GIF"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">citris</span><span class="p">,</span><span class="w"> 
  </span><span class="s2">"Vv GRF-GIF"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">vitis</span><span class="p">,</span><span class="w"> 
  </span><span class="s2">"Vv miR GRF-GIF"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">vitis_miR</span><span class="p">,</span><span class="w">
  </span><span class="s2">"Ta GRF-GIF"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">wheat</span><span class="p">)</span><span class="w">
</span></code></pre></div></div> <p><br></p> <h2 id="visualization">Visualization</h2> <p>For the Venn diagram visualisation, I used the R package <a href="https://github.com/yanlinlin82/ggvenn" rel="external nofollow noopener" target="_blank"><strong>ggvenn</strong></a>. If you haven’t already installed the <strong>ggvenn</strong> package, you can do so with the following commands:</p> <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Install package</span><span class="w">
</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">require</span><span class="p">(</span><span class="n">devtools</span><span class="p">))</span><span class="w"> </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"devtools"</span><span class="p">)</span><span class="w">
</span><span class="n">devtools</span><span class="o">::</span><span class="n">install_github</span><span class="p">(</span><span class="s2">"yanlinlin82/ggvenn"</span><span class="p">)</span><span class="w">

</span><span class="c1"># Load library</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggvenn</span><span class="p">)</span><span class="w">
</span></code></pre></div></div> <p>I created a 4-way Venn diagram using custom color coding for each GRF-GIF line, making it easy to discern the contribution of each set. I also customised the plot by decreasing the line width, increasing the font size for the name of each line, and disabling the percentage visualisation.</p> <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Plot 4-way venn diagram</span><span class="w">
</span><span class="n">ggvenn</span><span class="p">(</span><span class="w">
  </span><span class="n">venndata</span><span class="p">,</span><span class="w"> 
  </span><span class="n">fill_color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"#0073C2FF"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#EFC000FF"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#868686FF"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#CD534CFF"</span><span class="p">),</span><span class="w">
  </span><span class="n">stroke_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">set_name_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">show_percentage</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="w">
</span><span class="p">)</span><span class="w">
</span></code></pre></div></div> <p>The resulting diagram (below) provides a clear visual representation of shared and unique gene sets, aiding in the identification of conserved and species-specific responses to GRF-GIF expression.</p> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/venn_diagram-480.webp 480w,/assets/img/venn_diagram-800.webp 800w,/assets/img/venn_diagram-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/venn_diagram.jpg" class="img-fluid" width="100%" height="auto" title="4-way Venn diagram" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <div class="caption"> 4-way Venn diagram showing the overlap of DEGs across citrus, grape, grape with miRNA perturbation, and wheat GRF-GIF lines in <i>F. vesca</i>. </div> <p><br></p> <h2 id="key-insights">Key Insights</h2> <p>Beyond visualisation, I used the <strong>gplots</strong> library to extract a table of the overlapping gene names for downstream analysis.</p> <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Extract overlapping genes</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">gplots</span><span class="p">)</span><span class="w">
</span><span class="n">v.table</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">venn</span><span class="p">(</span><span class="n">venndata</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">v.table</span><span class="p">)</span><span class="w">
</span></code></pre></div></div> <p>By exporting and analysing the intersection of gene sets, we were able to identify clusters of genes that were consistently regulated across certain groups of lines. These conserved gene clusters may represent fundamental mechanisms by which GRF-GIF chimeras regulate growth and development across species, providing a basis for future functional studies.</p> <p><br></p> <h2 id="summary">Summary</h2> <p>A 4-way Venn diagram is a powerful tool to explore cross-sample gene expression. It underscores the utility of comparative transcriptomics in understanding conserved genetic responses and tailoring biotechnological interventions.</p> <p>If you’re interested in learning more about our study, you can read the full paper <a href="https://plantmethods.biomedcentral.com/articles/10.1186/s13007-024-01270-8" rel="external nofollow noopener" target="_blank"><strong>here</strong></a>. Otherwise, feel free to adapt the visualisation for your own datasets and explore the intersections within your gene sets.</p> </div> </article> <br> <hr> <br> <ul class="list-disc pl-8"></ul> <h2 class="text-3xl font-semibold mb-4 mt-12">Enjoy Reading This Article?</h2> <p class="mb-2">Here are some more articles you might like to read next:</p> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/blast/">Getting Started with BLAST on the Command Line</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/dotfiles/">Useful Bash Aliases and Functions</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/bash/">Getting Started with the Bash Command Line</a> </li> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 R. Jordan Price. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?5d75c11f89cd96294bf5e6dd1ee1bb30"></script> <script defer src="/assets/js/common.js?fcfacfb8c6281f5e68d5a7d348186eb1"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>