<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Visualising Gene Expression with a Heatmap using Python | Dr. R. Jordan Price </title> <meta name="author" content="R. Jordan Price"> <meta name="description" content="A guide to plotting a heatmap of differentially expressed genes (DEGs) in Python."> <meta name="keywords" content="bioinformatics, bioinformatician, computational-biology, molecular-biology, functional-genomics, academic-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://rj-price.github.io/blog/2024/heatmap/"> <script src="/assets/js/theme.js?a5ca4084d3b81624bcfa01156dae2b8e"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> Dr. R. Jordan Price </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/news/">news </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/apps/">apps </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Visualising Gene Expression with a Heatmap using Python</h1> <p class="post-meta"> December 16, 2024 </p> <p class="post-tags"> <a href="/blog/2024"> <i class="fa-solid fa-calendar fa-sm"></i> 2024 </a>   ·   <a href="/blog/tag/bioinformatics"> <i class="fa-solid fa-hashtag fa-sm"></i> bioinformatics</a>   <a href="/blog/tag/rnaseq"> <i class="fa-solid fa-hashtag fa-sm"></i> rnaseq</a>   <a href="/blog/tag/visualisation"> <i class="fa-solid fa-hashtag fa-sm"></i> visualisation</a>   <a href="/blog/tag/heatmap"> <i class="fa-solid fa-hashtag fa-sm"></i> heatmap</a>   </p> </header> <article class="post-content"> <div id="markdown-content"> <p>Heatmaps are essential tools for exploring gene expression data, allowing us to visualise patterns across samples or conditions, revealing relationships and highlighting key trends.</p> <p>In our recent study published in <a href="https://plantmethods.biomedcentral.com/articles/10.1186/s13007-024-01270-8" rel="external nofollow noopener" target="_blank">Plant Methods</a>, we used a heatmap to explore the expression of differentially expressed genes (DEGs) shared across four GRF-GIF chimera lines in the strawberry model organism, <em>Fragaria vesca</em>.</p> <p>In this post, I’ll explain how I generated the heatmap  (seen in <a href="https://plantmethods.biomedcentral.com/articles/10.1186/s13007-024-01270-8/figures/4" rel="external nofollow noopener" target="_blank">Figure 4b</a>), which displays the expression profiles of shared DEGs identified in the <a href="https://rjprice.bio/blog/2024/venn/" rel="external nofollow noopener" target="_blank">4-way Venn diagram</a>, using a <a href="https://github.com/rj-price/h4_grf-gif_rna/blob/main/scripts/Shared%20DEGs%20Heatmap.ipynb" rel="external nofollow noopener" target="_blank">Python and Jupyter Notebook workflow</a>.</p> <p><br></p> <h2 id="step-1-setting-up-the-environment">Step 1: Setting Up the Environment</h2> <p>First, the following Python libraries were imported:</p> <ul> <li> <strong>numpy</strong> and <strong>pandas</strong> were used to import data and extract shared DEGs.</li> <li> <strong>matplotlib</strong> and <strong>seaborn</strong> were used to generate the heatmap.</li> <li> <strong>scipy.cluster.hierarchy</strong> was used to determine and reorientate sample clustering.</li> </ul> <div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="n">pandas</span> <span class="k">as</span> <span class="n">pd</span>
<span class="kn">import</span> <span class="n">numpy</span> <span class="k">as</span> <span class="n">np</span>
<span class="kn">import</span> <span class="n">seaborn</span> <span class="k">as</span> <span class="n">sns</span>
<span class="kn">import</span> <span class="n">matplotlib.pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">from</span> <span class="n">scipy.cluster.hierarchy</span> <span class="kn">import</span> <span class="n">linkage</span><span class="p">,</span> <span class="n">dendrogram</span>
</code></pre></div></div> <p><br></p> <h2 id="step-2-gather-data">Step 2: Gather Data</h2> <p>Our analysis focussed on DEGs from <em>F. vesca</em> lines transformed with GRF-GIF chimeras from citrus (Cc GRF-GIF), grape (Vv GRF-GIF), grape with microRNA perturbation (Vv miR GRF-GIF), and wheat (Ta GRF-GIF).</p> <p>Following differential expression analysis, I defined variables for each output file containing the log2 fold-change (log2FC) values, as well as the list of shared DEGs identified using a <a href="https://rjprice.bio/blog/2024/venn/" rel="external nofollow noopener" target="_blank">4-way Venn diagram</a>.</p> <div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">condition1_file</span> <span class="o">=</span> <span class="sh">"</span><span class="s">GRF-GIF_Citris_vs_Control.tsv</span><span class="sh">"</span>
<span class="n">condition2_file</span> <span class="o">=</span> <span class="sh">"</span><span class="s">GRF-GIF_Vitis_miR_vs_Control.tsv</span><span class="sh">"</span>
<span class="n">condition3_file</span> <span class="o">=</span> <span class="sh">"</span><span class="s">GRF-GIF_Vitis_vs_Control.tsv</span><span class="sh">"</span>
<span class="n">condition4_file</span> <span class="o">=</span> <span class="sh">"</span><span class="s">GRF-GIF_Wheat_vs_Control.tsv</span><span class="sh">"</span>
<span class="n">subset_file</span> <span class="o">=</span> <span class="sh">"</span><span class="s">shared_DEG_names.txt</span><span class="sh">"</span>
</code></pre></div></div> <p><br></p> <h2 id="step-3-extracting-and-organising-data">Step 3: Extracting and Organising Data</h2> <p>Next, I loaded the shared DEG names and iterated through the condition files to extract log2FC values for these genes. The extracted data was compiled into a single dataframe, where rows represented genes and columns represented conditions.</p> <div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Create a set of shared DEG identifiers
</span><span class="k">with</span> <span class="nf">open</span><span class="p">(</span><span class="n">subset_file</span><span class="p">,</span> <span class="sh">"</span><span class="s">r</span><span class="sh">"</span><span class="p">)</span> <span class="k">as</span> <span class="n">subset</span><span class="p">:</span>
    <span class="n">subset_genes</span> <span class="o">=</span> <span class="nf">set</span><span class="p">(</span><span class="n">line</span><span class="p">.</span><span class="nf">strip</span><span class="p">()</span> <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">subset</span><span class="p">)</span>

<span class="c1"># Define empty output dataframe
</span><span class="n">output_df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nc">DataFrame</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="sh">"</span><span class="s">GeneID</span><span class="sh">"</span><span class="p">])</span>

<span class="c1"># Extract the log2FC values for shared DEGs and merge data across conditions
</span><span class="k">for</span> <span class="n">condition_file</span><span class="p">,</span> <span class="n">condition_name</span> <span class="ow">in</span> <span class="nf">zip</span><span class="p">(</span>
    <span class="p">[</span><span class="n">condition1_file</span><span class="p">,</span> <span class="n">condition2_file</span><span class="p">,</span> <span class="n">condition3_file</span><span class="p">,</span> <span class="n">condition4_file</span><span class="p">],</span>
    <span class="p">[</span><span class="sh">"</span><span class="s">Cc GRF-GIF</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">Vv miR GRF-GIF</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">Vv GRF-GIF</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">Ta GRF-GIF</span><span class="sh">"</span><span class="p">]</span>
<span class="p">):</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">read_csv</span><span class="p">(</span><span class="n">condition_file</span><span class="p">,</span> <span class="n">delim_whitespace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">subset_df</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="sh">"</span><span class="s">GeneID</span><span class="sh">"</span><span class="p">].</span><span class="nf">isin</span><span class="p">(</span><span class="n">subset_genes</span><span class="p">)]</span>
    <span class="n">subset_df</span> <span class="o">=</span> <span class="n">subset_df</span><span class="p">[[</span><span class="sh">"</span><span class="s">GeneID</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">log2FC</span><span class="sh">"</span><span class="p">]]</span>
    <span class="n">subset_df</span><span class="p">.</span><span class="nf">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="sh">"</span><span class="s">log2FC</span><span class="sh">"</span><span class="p">:</span> <span class="n">condition_name</span><span class="p">},</span> <span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">output_df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">merge</span><span class="p">(</span><span class="n">output_df</span><span class="p">,</span> <span class="n">subset_df</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="sh">"</span><span class="s">GeneID</span><span class="sh">"</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="sh">"</span><span class="s">outer</span><span class="sh">"</span><span class="p">)</span>

<span class="c1"># Reorder columns and set GeneID as index
</span><span class="n">output_df</span> <span class="o">=</span> <span class="n">output_df</span><span class="p">[[</span><span class="sh">"</span><span class="s">GeneID</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">Cc GRF-GIF</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">Vv GRF-GIF</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">Vv miR GRF-GIF</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">Ta GRF-GIF</span><span class="sh">"</span><span class="p">]]</span>
<span class="n">output_df</span><span class="p">.</span><span class="nf">set_index</span><span class="p">(</span><span class="sh">"</span><span class="s">GeneID</span><span class="sh">"</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
</code></pre></div></div> <p><br></p> <h2 id="step-4-sample-hierarchy-visualisation">Step 4: Sample Hierarchy Visualisation</h2> <p>To better understand the relationships between conditions, I performed hierarchical clustering on the samples. This clustering helps to visualise how similar or distinct the conditions are based on DEG expression. It also allowed me to reorientate the sample order for consistent visualisation across all panels of the figure.</p> <div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Use hierarchical clustering to establish sample relationships
</span><span class="n">link</span> <span class="o">=</span> <span class="nf">linkage</span><span class="p">(</span><span class="n">output_df</span><span class="p">.</span><span class="n">T</span><span class="p">)</span>
<span class="n">_</span> <span class="o">=</span> <span class="nf">dendrogram</span><span class="p">(</span><span class="n">link</span><span class="p">)</span>

<span class="c1"># Adjust sample order for intuitive visualisation
</span><span class="n">link</span><span class="p">[</span><span class="mi">2</span><span class="p">][[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">]]</span> <span class="o">=</span> <span class="n">link</span><span class="p">[</span><span class="mi">2</span><span class="p">][[</span><span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">]]</span>
<span class="n">_</span> <span class="o">=</span> <span class="nf">dendrogram</span><span class="p">(</span><span class="n">link</span><span class="p">)</span>
</code></pre></div></div> <p><br></p> <h2 id="step-5-generating-the-heatmap">Step 5: Generating the Heatmap</h2> <p>Finally, I visualised the shared DEG expression profiles using a clustered heatmap. The heatmap provides insights into expression patterns, highlighting upregulated and downregulated clusters.</p> <p><strong>Key parameters:</strong></p> <ul> <li> <strong><code class="language-plaintext highlighter-rouge">cmap="vlag"</code>:</strong> Divergent color scheme for highlighting up/downregulation.</li> <li> <strong><code class="language-plaintext highlighter-rouge">center=0</code>:</strong> Center the color scale at zero to distinguish positive and negative log2FC values.</li> <li> <strong><code class="language-plaintext highlighter-rouge">dendrogram_ratio</code>:</strong> Adjust the ratio of dendrogram to heatmap size for readability.</li> <li> <strong><code class="language-plaintext highlighter-rouge">row_cluster=True</code></strong>: Cluster genes to reveal groups with similar expression patterns.</li> </ul> <div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sns</span><span class="p">.</span><span class="nf">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.1</span><span class="p">)</span>
<span class="n">shared_heatmap</span> <span class="o">=</span> <span class="n">sns</span><span class="p">.</span><span class="nf">clustermap</span><span class="p">(</span>
    <span class="n">output_df</span><span class="p">,</span> <span class="n">center</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="sh">"</span><span class="s">vlag</span><span class="sh">"</span><span class="p">,</span> <span class="n">vmin</span><span class="o">=-</span><span class="mi">3</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
    <span class="n">dendrogram_ratio</span><span class="o">=</span><span class="p">(.</span><span class="mi">1</span><span class="p">,</span> <span class="p">.</span><span class="mi">05</span><span class="p">),</span>
    <span class="n">row_cluster</span> <span class="o">=</span> <span class="bp">True</span><span class="p">,</span> <span class="n">col_linkage</span><span class="o">=</span><span class="n">link</span><span class="p">,</span>
    <span class="n">cbar_pos</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mf">0.05</span><span class="p">,</span> <span class="p">.</span><span class="mi">32</span><span class="p">,</span> <span class="p">.</span><span class="mi">03</span><span class="p">,</span> <span class="p">.</span><span class="mi">2</span><span class="p">),</span>
    <span class="n">linewidths</span><span class="o">=</span><span class="p">.</span><span class="mi">75</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">20</span><span class="p">)</span>
<span class="p">)</span>

<span class="n">shared_heatmap</span><span class="p">.</span><span class="n">cax</span><span class="p">.</span><span class="nf">tick_params</span><span class="p">(</span><span class="n">labelsize</span><span class="o">=</span><span class="mi">18</span><span class="p">)</span>
<span class="n">shared_heatmap</span><span class="p">.</span><span class="n">ax_heatmap</span><span class="p">.</span><span class="nf">set_ylabel</span><span class="p">(</span><span class="sh">"</span><span class="s">GeneID</span><span class="sh">"</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">18</span><span class="p">)</span>
<span class="n">shared_heatmap</span><span class="p">.</span><span class="n">ax_heatmap</span><span class="p">.</span><span class="nf">set_xticklabels</span><span class="p">(</span><span class="n">shared_heatmap</span><span class="p">.</span><span class="n">ax_heatmap</span><span class="p">.</span><span class="nf">get_xticklabels</span><span class="p">(),</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">18</span><span class="p">)</span>

<span class="n">shared_heatmap</span><span class="p">.</span><span class="nf">savefig</span><span class="p">(</span><span class="sh">'</span><span class="s">Shared DEGs Heatmap.tif</span><span class="sh">'</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="sh">"</span><span class="s">tight</span><span class="sh">"</span><span class="p">)</span>
</code></pre></div></div> <p><br></p> <h2 id="results">Results</h2> <p>The resulting heatmap revealed distinct clusters of genes with differentially shared expression across the four GRF-GIF lines. This visualisation provided a critical bridge between the <a href="https://rjprice.bio/blog/2024/venn/" rel="external nofollow noopener" target="_blank">4-way Venn analysis</a> and functional characterisation of shared DEGs.</p> <p><br></p> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/heatmap-480.webp 480w,/assets/img/heatmap-800.webp 800w,/assets/img/heatmap-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/heatmap.jpg" class="img-fluid" width="100%" height="auto" title="4-way Venn diagram" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <div class="caption"> Heatmap showing the log2FC expression of shared DEGs across citrus, grape, grape with miRNA perturbation, and wheat GRF-GIF lines in <i>F. vesca</i>. </div> <p><br></p> <h2 id="summary">Summary</h2> <p>This step-by-step approach demonstrates how Python can streamline the process of DEG visualisation, from raw data to polished heatmaps. The insights gained through this analysis helped drive our understanding of GRF-GIF-mediated gene regulation.</p> <p>Explore the full notebook and other code from our paper in my <a href="https://github.com/rj-price/h4_grf-gif_rna" rel="external nofollow noopener" target="_blank">GitHub repository</a>.</p> <p><br></p> <h2 id="support">Support</h2> <p>If you find this blog post helpful, please consider <a href="https://ko-fi.com/jordanprice" rel="external nofollow noopener" target="_blank">buying me a coffee on Ko-fi</a>. Your support is very much appreciated!</p> <p style="text-align: center"> <a href="https://ko-fi.com/jordanprice" target="_blank" rel="external nofollow noopener"> <img height="36" style="border:0px;height:36px;" src="https://storage.ko-fi.com/cdn/kofi2.png?v=3" border="0" alt="Buy Me a Coffee at ko-fi.com"> </a> </p> </div> </article> <br> <hr> <br> <ul class="list-disc pl-8"></ul> <h2 class="text-3xl font-semibold mb-4 mt-12">Enjoy Reading This Article?</h2> <p class="mb-2">Here are some more articles you might like to read next:</p> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/venn/">4-Way Venn Diagram of Overlapping Gene Expression in R</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2025/eda_app/">A Streamlit App for Rapid Exploratory Data Analysis and Visualisation</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2025/snp_tree/">Reconstructing a Phylogenetic Tree Using SNP Information from WGS Data</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/jellyfish/">Estimating Genome Size Using Jellyfish and GenomeScope2</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/blast/">Getting Started with BLAST on the Command Line</a> </li> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 R. Jordan Price. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?5d75c11f89cd96294bf5e6dd1ee1bb30"></script> <script defer src="/assets/js/common.js?fcfacfb8c6281f5e68d5a7d348186eb1"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>