<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Reconstructing a Phylogenetic Tree Using SNP Information from WGS Data | Dr. R. Jordan Price </title> <meta name="author" content="R. Jordan Price"> <meta name="description" content="A guide to generating a phylogenetic tree from whole genome sequencing data."> <meta name="keywords" content="bioinformatics, bioinformatician, computational-biology, molecular-biology, functional-genomics, academic-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%A7%AC&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://rj-price.github.io/blog/2025/snp_tree/"> <script src="/assets/js/theme.js?a5ca4084d3b81624bcfa01156dae2b8e"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> Dr. R. Jordan Price </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/news/">news </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Reconstructing a Phylogenetic Tree Using SNP Information from WGS Data</h1> <p class="post-meta"> January 21, 2025 </p> <p class="post-tags"> <a href="/blog/2025"> <i class="fa-solid fa-calendar fa-sm"></i> 2025 </a>   ·   <a href="/blog/tag/bioinformatics"> <i class="fa-solid fa-hashtag fa-sm"></i> bioinformatics</a>   <a href="/blog/tag/genomics"> <i class="fa-solid fa-hashtag fa-sm"></i> genomics</a>   <a href="/blog/tag/visualisation"> <i class="fa-solid fa-hashtag fa-sm"></i> visualisation</a>   <a href="/blog/tag/phylogeny"> <i class="fa-solid fa-hashtag fa-sm"></i> phylogeny</a>     ·   <a href="/blog/category/getting-started"> <i class="fa-solid fa-tag fa-sm"></i> getting-started</a>   </p> </header> <article class="post-content"> <div id="markdown-content"> <p>Phylogenetic analysis is a key part of evolutionary biology, allowing researchers to infer the relationships between species, populations, or strains.</p> <p>By using whole genome sequencing (WGS) data, we can identify single nucleotide polymorphisms (SNPs) and use them to construct high resolution phylogenetic trees.</p> <p>In this blog post, I’ll guide you through reconstructing the phylogenetic tree of <em>Rhynchosporium commune</em>, a significant fungal pathogen of barley crops, using SNP information obtained from WGS data. This tutorial provides a practical, step-by-step guide from raw data processing through to tree visualisation.</p> <p><br></p> <h2 id="environment-setup">Environment Setup</h2> <p>Before we start, we’ll need to set up the environment and install all of the required software.</p> <p>The scripts used in this tutorial can be found in my <a href="https://github.com/rj-price/rhynchosporium_tree" rel="external nofollow noopener" target="_blank">GitHub repository</a>, and are written to be run on a Slurm-based HPC cluster. If you have Conda installed, you can install all of the software in a new environment as below:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Clone repository</span>
git clone https://github.com/rj-price/rhynchosporium_tree.git
<span class="nb">cd </span>rhynchosporium_tree

<span class="c"># Create conda environment</span>
conda create <span class="nt">-n</span> snp_tree <span class="se">\</span>
	ncbi-datasets-cli <span class="se">\</span>
	fastqc <span class="se">\</span>
	multiqc <span class="se">\</span>
	trimmomatic <span class="se">\</span>
	python <span class="se">\</span>
	pandas <span class="se">\</span>
	biopython <span class="se">\</span>
	bwa <span class="se">\</span>
	samtools <span class="se">\</span>
	bcftools <span class="se">\</span>
	raxml-ng

<span class="c"># Activate conda environment</span>
conda activate snp_tree
</code></pre></div></div> <p><br></p> <h2 id="data-preparation">Data Preparation</h2> <h3 id="download-wgs-data-and-reference-genome">Download WGS Data and Reference Genome</h3> <p>With the environment set up, the next step involves acquiring the raw sequencing data and a reference genome.</p> <p>For this analysis, I searched the Sequence Read Archive (SRA) for <em>Rhynchosporium</em> WGS data and downloaded the available metadata (<code class="language-plaintext highlighter-rouge">SraRunTable.txt</code>).</p> <p>I then used pre-generated scripts via the European Nucleotide Archive (ENA) to downloaded all of the <em>Rhynchosporium</em> reads from the BioProject accessions <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5972619" rel="external nofollow noopener" target="_blank">PRJNA327656</a> and PRJNA419548:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Create directory for raw reads</span>
<span class="nb">mkdir </span>reads <span class="o">&amp;&amp;</span> <span class="nb">cd </span>reads

<span class="c"># Download WGS data using pre-generated scripts</span>
bash ../scripts/ena-file-download-selected-files-20240216-1125.sh
bash ../scripts/ena-file-download-selected-files-20240216-1138.sh
</code></pre></div></div> <p>For the reference genome, the NCBI Datasets tool was used to download the <em>R. commune</em> UK7 genome:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Create directory for reference genome</span>
<span class="nb">mkdir </span>UK7 <span class="o">&amp;&amp;</span> <span class="nb">cd </span>UK7

<span class="c"># Download UK7 genome</span>
datasets download genome accession GCA_900074885.1 <span class="nt">--filename</span> GCA_900074885.1.zip

<span class="c"># Unzip, move and rename</span>
unzip GCA_900074885.1.zip
<span class="nb">mv</span> <span class="se">\</span>
	ncbi_dataset/data/GCA_900074885.1/GCA_900074885.1_version_1_genomic.fna <span class="se">\</span>
	R_commune_UK7.fasta
</code></pre></div></div> <p><br></p> <h3 id="quality-control-of-sequencing-data">Quality Control of Sequencing Data</h3> <p>Downstream analyses rely on high quality sequencing data. To assess the quality of the raw reads, I used <strong>FastQC</strong>, followed by <strong>MultiQC</strong> to summarise the results:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Run FastQC on raw data</span>
<span class="nb">cd </span>reads
<span class="k">for </span>file <span class="k">in</span> <span class="k">*</span>gz<span class="p">;</span> <span class="k">do
    </span>sbatch ../scripts/fastqc.sh <span class="nv">$file</span>
<span class="k">done</span>

<span class="c"># Aggregate results with MultiQC</span>
multiqc fastqc/
</code></pre></div></div> <p>Adapters and low quality sequences were trimmed using <strong>Trimmomatic</strong> based on the QC report, followed by a second round of quality control:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Create directory for trimmed reads</span>
<span class="nb">mkdir </span>trimmed <span class="o">&amp;&amp;</span> <span class="nb">cd </span>trimmed

<span class="c"># Trim reads using Trimmomatic</span>
<span class="k">for </span>fwd <span class="k">in</span> ../reads/<span class="k">*</span>_1.fastq.gz<span class="p">;</span> <span class="k">do
    </span><span class="nv">rev</span><span class="o">=</span><span class="si">$(</span><span class="nb">ls</span> <span class="nv">$fwd</span> | <span class="nb">sed</span> <span class="s1">'s/_1/_2/g'</span><span class="si">)</span>
    sbatch ../scripts/trimmomatic_pe.sh <span class="nv">$fwd</span> <span class="nv">$rev</span>
<span class="k">done</span>

<span class="c"># QC on trimmed reads</span>
<span class="k">for </span>file <span class="k">in</span> <span class="k">*</span>gz<span class="p">;</span> <span class="k">do
    </span>sbatch ../scripts/fastqc.sh <span class="nv">$file</span>
<span class="k">done

</span>multiqc fastqc/
</code></pre></div></div> <p>Finally, to facilitate easy comparisons across samples, a custom python script was used to rename the reads using a custom naming convention based on the available SRA metadata:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">mkdir </span>renamed_reads
python scripts/rename_reads.py SraRunTable.txt trimmed/ renamed_reads/
</code></pre></div></div> <p><br></p> <h2 id="phylogenetic-tree-reconstruction">Phylogenetic Tree Reconstruction</h2> <h3 id="mapping-reads-to-the-reference-genome">Mapping Reads to the Reference Genome</h3> <p>To identify SNPs, the reads were aligned to the UK7 reference genome using <strong>BWA</strong>. After indexing the genome, the reads were mapped, and the alignments were sorted and indexed using <strong>Samtools</strong>:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Index the genome</span>
bwa index UK7/R_commune_UK7.fasta

<span class="c"># Map reads to the genome</span>
<span class="nb">mkdir </span>alignment <span class="o">&amp;&amp;</span> <span class="nb">cd </span>alignment
<span class="k">for </span>fwd <span class="k">in</span> ../renamed_reads/<span class="k">*</span>_F.fastq.gz<span class="p">;</span> <span class="k">do
    </span><span class="nv">rev</span><span class="o">=</span><span class="si">$(</span><span class="nb">ls</span> <span class="nv">$fwd</span> | <span class="nb">sed</span> <span class="s1">'s/_F/_R/g'</span><span class="si">)</span>
    sbatch ../scripts/bwa-mem.sh ../UK7/R_commune_UK7.fasta <span class="nv">$fwd</span> <span class="nv">$rev</span>
<span class="k">done</span>

<span class="c"># Sort and index alignments</span>
<span class="k">for </span>file <span class="k">in</span> <span class="k">*</span>sam<span class="p">;</span> <span class="k">do
    </span>sbatch ../scripts/samtools_sort.sh <span class="nv">$file</span>
<span class="k">done</span>
</code></pre></div></div> <p><br></p> <h3 id="variant-calling-and-filtering">Variant Calling and Filtering</h3> <p>After generating a pileup of aligned reads, variants were called using <strong>bcftools</strong>. These were then filtered for high-quality SNPs based on quality scores, retaining only the highest-confidence SNPs for phylogenetic analysis:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Index genome</span>
samtools faidx UK7/R_commune_UK7.fasta

<span class="c"># Generate pileup and call variants</span>
<span class="nb">mkdir </span>variant_calling <span class="o">&amp;&amp;</span> <span class="nb">cd </span>variant_calling
<span class="k">for </span>file <span class="k">in</span> ../alignment/<span class="k">*</span>bam<span class="p">;</span> <span class="k">do
    </span><span class="nb">realpath</span> <span class="nv">$file</span> <span class="o">&gt;&gt;</span> rhynchosporium.list
<span class="k">done
</span>sbatch ../scripts/bcftools_call.sh ../UK7/R_commune_UK7.fasta rhynchosporium.list

<span class="c"># Filter variants</span>
bcftools view <span class="se">\</span>
	rhynchosporium.vcf <span class="nt">-e</span> <span class="s1">'QUAL&lt;800'</span> <span class="nt">--types</span> snps <span class="se">\</span>
	<span class="o">&gt;</span> rhynchosporium_highqual.vcf
</code></pre></div></div> <p><br></p> <h3 id="preparing-data-for-phylogenetic-analysis">Preparing Data for Phylogenetic Analysis</h3> <p>The filtered SNP data in VCF format was converted to PHYLIP format using the <code class="language-plaintext highlighter-rouge">vcf2phylip.py</code> script from https://github.com/edgardomortiz/vcf2phylip and invariant sites were removed using the <code class="language-plaintext highlighter-rouge">ascbias.py</code> script from https://github.com/btmartin721/raxml_ascbias:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Convert VCF to PHYLIP</span>
python ../scripts/vcf2phylip.py <span class="nt">-i</span> ./rhynchosporium_highqual.vcf

<span class="c"># Remove invariant sites</span>
python ../scripts/ascbias.py <span class="nt">-p</span> ./rhynchosporium_highqual.min4.phy <span class="nt">-o</span> rhynchosporium_highqual.phy
</code></pre></div></div> <p><br></p> <h3 id="building-the-phylogenetic-tree">Building the Phylogenetic Tree</h3> <p>With the prepared dataset, I used <strong>RAxML-ng</strong> for phylogenetic inference:</p> <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sbatch ../scripts/raxml.sh rhynchosporium_highqual.phy
</code></pre></div></div> <p><br></p> <h2 id="visualising-the-tree">Visualising the Tree</h2> <p>The final phylogenetic tree was visualised using <a href="https://itol.embl.de/" rel="external nofollow noopener" target="_blank">iTOL</a>, a useful online tool for exploring and annotating trees. Below are examples of the visualisations generated:</p> <p><br></p> <div class="row"> <a href="assets/img/rhyncho_tree_branch.jpg" target="_blank"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/rhyncho_tree_branch-480.webp 480w,/assets/img/rhyncho_tree_branch-800.webp 800w,/assets/img/rhyncho_tree_branch-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/rhyncho_tree_branch.jpg" class="img-fluid" width="100%" height="auto" title="Phylogenetic Tree with Branch Distances" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </a> <a href="assets/img/rhyncho_tree_no-branch.jpg" target="_blank"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/rhyncho_tree_no-branch-480.webp 480w,/assets/img/rhyncho_tree_no-branch-800.webp 800w,/assets/img/rhyncho_tree_no-branch-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/rhyncho_tree_no-branch.jpg" class="img-fluid" width="100%" height="auto" title="Phylogenetic Tree without Branch Distances" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </a> </div> <div class="caption"> Phylogenetic tree of <i>Rhynchosporium commune</i> SNPs with and without branch lengths. Samples are coloured by country of origin. </div> <p><br></p> <h2 id="summary">Summary</h2> <p>Reconstructing a phylogenetic tree from WGS data is a multi-step process involving data preparation, SNP calling, and tree inference. This guide outlines the workflow for analysing <em>Rhynchosporium commune</em> WGS data but can be adapted for other species or datasets.</p> <p>By following these steps, you can uncover evolutionary insights and explore the genetic diversity of your own organisms of interest.</p> </div> </article> <br> <hr> <br> <ul class="list-disc pl-8"></ul> <h2 class="text-3xl font-semibold mb-4 mt-12">Enjoy Reading This Article?</h2> <p class="mb-2">Here are some more articles you might like to read next:</p> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/jellyfish/">Estimating Genome Size Using Jellyfish and GenomeScope2</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/blast/">Getting Started with BLAST on the Command Line</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/heatmap/">Visualising Gene Expression with a Heatmap using Python</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/dotfiles/">Useful Bash Aliases and Functions</a> </li> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/venn/">4-Way Venn Diagram of Overlapping Gene Expression in R</a> </li> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 R. Jordan Price. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?5d75c11f89cd96294bf5e6dd1ee1bb30"></script> <script defer src="/assets/js/common.js?fcfacfb8c6281f5e68d5a7d348186eb1"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>